docs/build_profiles.md

```diff
@@ -0,0 +1,19 @@
+# Build Profiles
+Mbed 5.0 supports three primary build profiles, *default*, *debug* and *small*. When using
+the online compiler the *default* profile is used. When building from the command line
+the desired profile can be can be selected by adding the ```--profile mbed-os/tools/<profile>``` command line flag.
```

Comment:
It would be nice if the 3 built-in profiles could be used without specifying the full path. This isn't the case currently though, so the docs should be updated in the PR that makes this change. In that case I would vote for keeping user provided profiles.